scripts/west_commands/runners/dediprog.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2017 Linaro Limited.
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Dediprog (dpcmd) flash only runner for SPI chips.'''
+
+import platform
+
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+DPCMD_EXE = 'dpcmd.exe' if platform.system() == 'Windows' else 'dpcmd'
+
+class DediProgBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for DediProg (dpcmd).'''
+
+    def __init__(self, cfg, spi_image, vcc):
+        super(DediProgBinaryRunner, self).__init__(cfg)
+        self.spi_image = spi_image
+        self.vcc = vcc
+
+    @classmethod
+    def name(cls):
+        return 'dediprog'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--spi-image', required=True,
+                            help='path to SPI image')
+        parser.add_argument('--vcc',
+                            help='VCC (0=3.5V, 1=2.5V, 2=1.8V)')
+
+    @classmethod
+    def create(cls, cfg, args):
+        return DediProgBinaryRunner(cfg,
+                                    spi_image=args.spi_image,
+                                    vcc=args.vcc)
+
+    def do_run(self, command, **kwargs):
+        self.require(DPCMD_EXE)
+        cmd_flash = [DPCMD_EXE, '-u', self.spi_image]
```

Comment:
Short options like `-u` are great to save time typing. In a script please prefer the longer and clearer version, here: `--auto`